controllers/toolchainclusterhealth/toolchaincluster_healthcheck_controller.go

```diff
@@ -0,0 +1,97 @@
+package toolchainclusterhealth
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	toolchainv1alpha1 "github.com/codeready-toolchain/api/api/v1alpha1"
+	"github.com/codeready-toolchain/toolchain-common/pkg/cluster"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeclientset "k8s.io/client-go/kubernetes"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// NewReconciler returns a new Reconciler
+func NewReconciler(mgr manager.Manager, namespace string, timeout time.Duration, requeAfter time.Duration) *Reconciler {
+	cacheLog := log.Log.WithName("toolchaincluster_health")
+	clusterCacheService := cluster.NewToolchainClusterService(mgr.GetClient(), cacheLog, namespace, timeout)
+	return &Reconciler{
+		client:              mgr.GetClient(),
+		scheme:              mgr.GetScheme(),
+		clusterCacheService: clusterCacheService,
+		requeAfter:          requeAfter,
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&toolchainv1alpha1.ToolchainCluster{}).
+		Complete(r)
+}
+
+// Reconciler reconciles a ToolchainCluster object
+type Reconciler struct {
+	client              client.Client
+	scheme              *runtime.Scheme
+	clusterCacheService cluster.ToolchainClusterService
```

Comment:
maybe I'm missing something but where are we using this?

Reply:
II am not sure if i understand the question right 
but the struct is being used here https://github.com/codeready-toolchain/toolchain-common/pull/386/files#diff-da3ed27ca5df992db5fa4b9114064a8fe6b6a3b4a4484a12cfa84590c6c248daR52 
and https://github.com/codeready-toolchain/toolchain-common/pull/386/files#diff-da3ed27ca5df992db5fa4b9114064a8fe6b6a3b4a4484a12cfa84590c6c248daR33 and 
https://github.com/codeready-toolchain/toolchain-common/pull/386/files#diff-da3ed27ca5df992db5fa4b9114064a8fe6b6a3b4a4484a12cfa84590c6c248daR21

Reply:
its not being used anywhere in the reconciler, l removed it now

Reply:
right, sorry for the confusion. I was referring to the `clusterCacheService` object indeed 👍